File: lib/domain/entities/support.dart
import 'package:equatable/equatable.dart';

/// Support ticket category
enum TicketCategory {
  schoolVerification('school_verification', '學校驗證'),
  payment('payment', '付款問題'),
  bugReport('bug_report', 'Bug 回報'),
  other('other', '其他');

  final String value;
  final String label;
  const TicketCategory(this.value, this.label);

  static TicketCategory fromString(String value) {
    return TicketCategory.values.firstWhere(
      (e) => e.value == value,
      orElse: () => TicketCategory.other,
    );
  }
}

/// Support ticket status
enum TicketStatus {
  open('open', '待處理'),
  inProgress('in_progress', '處理中'),
  resolved('resolved', '已解決'),
  closed('closed', '已關閉');

  final String value;
  final String label;
  const TicketStatus(this.value, this.label);

  static TicketStatus fromString(String value) {
    return TicketStatus.values.firstWhere(
      (e) => e.value == value,
      orElse: () => TicketStatus.open,
    );
  }

  bool get isClosed => this == resolved || this == closed;
}

/// Support ticket entity
class SupportTicket extends Equatable {
  final String id;
  final String userId;
  final TicketCategory category;
  final String subject;
  final TicketStatus status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DateTime? resolvedAt;

  const SupportTicket({
    required this.id,
    required this.userId,
    required this.category,
    required this.subject,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.resolvedAt,
  });

  @override
  List<Object?> get props => [id, userId, category, subject, status, createdAt, updatedAt, resolvedAt];
}

/// Support message entity
class SupportMessage extends Equatable {
  final String id;
  final String ticketId;
  final String senderType;
  final String senderId;
  final String? content;
  final String? imagePath;
  final String? imageUrl;
  final DateTime createdAt;

  const SupportMessage({
    required this.id,
    required this.ticketId,
    required this.senderType,
    required this.senderId,
    this.content,
    this.imagePath,
    this.imageUrl,
    required this.createdAt,
  });

  bool get isUser => senderType == 'user';
  bool get isAdmin => senderType == 'admin';
  bool get hasImage => imagePath != null && imagePath!.isNotEmpty;

  @override
  List<Object?> get props => [id, ticketId, senderType, senderId, content, imagePath, createdAt];
}

File: admin/src/types/support.ts
export type TicketCategory = 'school_verification' | 'payment' | 'bug_report' | 'other'
export type TicketStatus = 'open' | 'in_progress' | 'resolved' | 'closed'
export type SenderType = 'user' | 'admin'

export interface SupportTicket {
  id: string
  user_id: string
  category: TicketCategory
  subject: string
  status: TicketStatus
  created_at: string
  updated_at: string
  resolved_at: string | null
}

export interface SupportMessage {
  id: string
  ticket_id: string
  sender_type: SenderType
  sender_id: string
  content: string | null
  image_path: string | null
  created_at: string
}

export interface TicketWithUser extends SupportTicket {
  user_nickname: string | null
  user_gender: string | null
  university_name: string | null
}

export const TICKET_CATEGORY_LABELS: Record<TicketCategory, string> = {
  school_verification: '學校驗證',
  payment: '付款問題',
  bug_report: 'Bug 回報',
  other: '其他',
}

export const TICKET_STATUS_LABELS: Record<TicketStatus, string> = {
  open: '待處理',
  in_progress: '處理中',
  resolved: '已解決',
  closed: '已關閉',
}

export const TICKET_STATUS_COLORS: Record<TicketStatus, string> = {
  open: 'bg-amber-100 text-amber-800',
  in_progress: 'bg-blue-100 text-blue-800',
  resolved: 'bg-emerald-100 text-emerald-800',
  closed: 'bg-gray-100 text-gray-500',
}
